src/snowflake/connector/pandas_tools.py

```diff
@@ -170,6 +172,15 @@ def write_pandas(
     if chunk_size is None:
         chunk_size = len(df)
 
+    if not isinstance(df.index, pandas.RangeIndex) or df.index.step != 1:
```

Comment:
hmmm if df.index.start is not 0, it will be written?

Reply:
are you suggesting adding a check that df.index.start must be 0 as well?

Reply:
Yes, if range index is 2, 3, 4, ... 10, it will also become 1, 2, ..., 9 after pulling data out as a pandas dataframe, right?

Reply:
we can add that and be even more restrictive, i.e. only allowing the standard range-index (which is step==1 and start==0). However, I opted to issue a warning at this point to allow code to still run. Basically the semantics are:
1.) issue warning if not standard pandas index
2.) load/ingest without index

Does that sound reasonable?

Reply:
Yes, sounds good. We should do that to avoid behavior change

Reply:
fixed.